source/jail.cpp

```diff
@@ -169,7 +169,7 @@ void CJailCell::PeriodicCheck( void )
 {
 	time_t now;
 	time( &now );
-	for( size_t i = playersInJail.size() - 1; i >= 0 && i < playersInJail.size(); --i )
+	for( size_t i = playersInJail.size() - 1; i > 0 && i < playersInJail.size(); --i )
```

Comment:
Warning below for reference:
```
jail.cpp: In member function ‘void JailCell::PeriodicCheck()’:
jail.cpp:174:46: warning: comparison of unsigned expression in ‘>= 0’ is always true [-Wtype-limits]
  174 |  for( size_t i = playersInJail.size() - 1; i >= 0 && i < playersInJail.size(); --i )```

Reply:
For sure.  I suspect this is relying on wrap around behaviour.  But if we don't restore the >=, you'll miss the first element.

The other option, of course, is something more like:
```suggestion
	for( size_t i = playersInJail.size(); i > 0; --i ) {
	  // Do work here addressing array as playersInJail[i - 1]
	}
```

Reply:
The pushed change doesn't resolve this - now we have a different problem.  Rather than first index being skipped, it's skipping the last.

See this sample code:
```
  size_t size = 5;
  for (size_t i = size - 1; i-- > 0 && i < size; ) {
    std::cout << "Hello instance " << i << std::endl;
  }
```
That's the same as your pattern - but it only iterates 4 times, not 5 as would be expected.  If you want to avoid border wrapping, you need a pattern like:
```
  for (size_t idx = size; idx > 0; --idx) {
    size_t i = idx - 1;
    std::cout << "Hello instance " << i << std::endl;
  }
```

Reply:
I forgot to remove the `- 1` for the 3 `for` loops. The following loop iterates the expected 32 times.

```c
for(size_t i = 32; i-- > 0; )
  printf("%d\n", i);
```

Output:
```
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1
0
```

Reply:
Fixed in b4721f4.